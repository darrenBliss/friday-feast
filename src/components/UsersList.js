// @flow
import * as React from 'react'
import styled from 'styled-components'
import { Card } from '../style/Card'

const ListHolder = styled.ul`
  margin: 0;
`

const List = styled.li`
  background-color: ${props => props.isActive ? 'papayawhip' : '#fff'};
  padding: .5rem;
`

type Props = {
  users: Array<Object>,
  activeUserIndex: number
}

const UsersList = ({ users, activeUserIndex }: Props) => {
  const userList = users.map((item, index) => {
    const isActive = activeUserIndex === index ? true : false

    return (
      <List key={item.uid} isActive={isActive}>
        {item.name}
      </List>
    )
  })

  return (
    <Card>
      <ListHolder>
        {userList}
      </ListHolder>
    </Card>
  )
}

export default UsersList