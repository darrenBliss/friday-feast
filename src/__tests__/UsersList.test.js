import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import UsersList from '../components/UsersList'

const users = [{
  name: "Megatron",
  uid: 0
}, {
  name: "Totoro",
  uid: 1
}]

describe('UserItem component', () => {
  it('Render snapshot', () => {
    const wrapper = shallow(<UsersList users={users} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Render snapshot', () => {
    const wrapper = shallow(<UsersList users={users} activeUserIndex={1} />)

    expect(wrapper.find('UsersList__List').at(1).props().isActive).toEqual(true)
  })
})
