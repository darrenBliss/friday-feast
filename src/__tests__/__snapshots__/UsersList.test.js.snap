// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem component Render snapshot 1`] = `
<UsersList__ListHolder>
  <UsersList__List
    isActive={false}
    key="0"
  >
    Megatron
  </UsersList__List>
  <UsersList__List
    isActive={false}
    key="1"
  >
    Totoro
  </UsersList__List>
</UsersList__ListHolder>
`;
